### DISCUSSION TITLE
Question About index_file_size and Index params

### DISCUSSION THREAD

**Discussion by elk-cloner** [milvus-io/milvus#6793] [Q&A and General discussion] [answered]
Hi, I read some documents of milvus but can't figure out my answer to these questions:
If I understand correctly, milvus keeps inserting vectors to a segment until index_file_size hits. So after index_file_size hits milvus starts building an index, my questions are that 

1. if we already have some segment with index_file_size size, does milvus build a new index for this new segment?
2. I mean at the end do we have two segments and two separate indexes ?
3. or milvus merges all segments and builds one index containing all entities of two segments ?
4. If it merges all segments to build new index it means it does training phase for index each time it wants to build new index(merge data) ?
5. If it keeps two separate indexes, does index params apply to each index ?  For example, if I use IVF_FLAT it’s important to know the number of vectors to set nlist arguments properly. 
6. If it keeps two separate indexes, does it search each index separately and then merge the result ?

**Comment by elk-cloner** [comment]
sry I have found my answers from these link:
1. https://milvus.io/docs/v2.0.0/index.md#Index-by-segment
2. https://milvus.io/docs/v2.0.0/glossary.md#Segment

**Comment by yhmo** [comment, marked as answer]
The "index_file_size" is a parameter of v1.x. The v2.0 doesn't have this parameter. In the v2.0, segment size is defined by a configuration in data_node.yaml, please refer the documentation: https://milvus.io/docs/v2.0.0/configuration_standalone-advanced.md
```
datacoord:
  segment:
    maxSize: 512 # MB
```

The main logic to build index is similar between v1.x and v2.0.
- When a segment size is less than a threshold, the segment is "growing", it can be merged.
- Once a segment size is exceed a threshold, the segment become "sealed", it cannot be merged any more. Only can be build index.
- Each index is independent. All index using same index parameters. But the "nlist" may be changed internally if a segment row count is to few.
- Search request is executed in each index, with same search parameters. Then merge the result to a final result return to client.

**Reply by elk-cloner** [reply]
thanks , you've made it crystal clear for me 👍
